discovery/gossiper_test.go

```diff
@@ -202,6 +202,27 @@ func (r *mockGraphSource) ForEachNode(func(node *channeldb.LightningNode) error)
 
 func (r *mockGraphSource) ForAllOutgoingChannels(cb func(i *channeldb.ChannelEdgeInfo,
 	c *channeldb.ChannelEdgePolicy) error) error {
+
+	chans := make(map[uint64]channeldb.ChannelEdge)
+	for _, info := range r.infos {
+		info := info
```

Comment:
Nit: why is this necessary?

Reply:
If we don't re-bind to the loop variable, then the pointer below will end up pointing to the same channel within the map.

Reply:
There's no pointer 😛